### ISSUE TITLE
Which version of Ruby to use?

### ISSUE THREAD

**Issue by lurch** [mraible/infoq-mini-book#24]
Sorry if this is a dumb question, but I'm entirely new to the Ruby ecosystem...

The README in this project says:
```
To be able to generate the PDF, you need have some dependencies installed.

On macOS, run the following commands:

rvm install "ruby-2.3.1"
gem install bundler -v '< 2'
bundle install
```

I'm running on Ubuntu 14.04 and so I did the 3 lines above, and then `./generate-pdf.sh` errors with:
```
Could not find Ascii85-1.0.3 in any of the sources
Run `bundle install` to install missing gems.
Could not find Ascii85-1.0.3 in any of the sources
Run `bundle install` to install missing gems.
```
So then I tried switching my "default Ruby" (?) with `rvm use ruby-2.3.1` and running the above three commands again, but now `bundle install` errors with:
```
Fetching gem metadata from https://rubygems.org/.........
geom2d-0.2.0 requires ruby version >= 2.4, which is incompatible with the
current version, ruby 2.3.1p112
```
And after this I've got no idea what to try next!!

**Comment by mraible** [comment]
It seems like you're saying [this commit](https://github.com/mraible/infoq-mini-book/commit/68ed8dbde5cebbac9d7a3b106dc5d5ab6cef92c2) messed things up. Can you try adding `gem 'Ascii85'` back into `Gemfile` and see if that fixes it?

**Comment by lurch** [comment]
I added `gem 'Ascii85'` to the bottom of `Gemfile` and then did:
```
rvm use ruby-2.6
bundle install
./generate-pdf.sh 
```
and got the same `Could not find Ascii85-1.0.3 in any of the sources` error.
And then I looked at the source of `generate-pdf.sh` and saw that it was doing `rvm use 2.3.1 --quiet`. If I comment out that line then `./generate-pdf.sh` now appears to be running successfully!

On a sidenote, the README says:
"Open build/asciidoc/pdf-pdf-screen/index.pdf to see the downloadable 8.5x11" PDF file." - but on my system it seems to end up at `build/asciidoc/pdf-screen/index.pdf` ?

**Comment by mojavelinux** [comment]
Please don't add the Ascii85 gem. That should never have gotten into the dependency tree. It has no relationship to this project.

**Comment by mojavelinux** [comment]
> And then I looked at the source of generate-pdf.sh and saw that it was doing rvm use 2.3.1 --quiet

When I included that in the script, I was trying to be too helpful / clever. Although it had the potential of removing the step to active RVM, it can also go wrong.

I think it would be better if the instructions said:

1. Install RVM
2. Install Ruby using RVM: `rvm install 2.6` if you haven't already
3. Activate Ruby using RVM: `rvm use 2.6`
4. Install bundler: `gem install bundler`
4. Run the script: ./generate-pdf.sh

And we should take out the two RVM commands at the top of the script. (In fact, that would allow the script to work even if you aren't using RVM).

**Comment by mojavelinux** [comment]
> but on my system it seems to end up at build/asciidoc/pdf-screen/index.pdf ?

that's correct

**Comment by mojavelinux** [comment]
> [Ascii85] has no relationship to this project.

Technically, it's pulled in by pdf-reader, but it doesn't have a direct relationship to this project.

**Comment by mojavelinux** [comment]
I sent a PR to clean things up.

**Comment by mraible** [comment]
Hopefully, this is fixed now. If not, please re-open.

**Comment by lurch** [comment]
This will be fixed once #26 is merged... :wink:
